paramiko/pipe.py

```diff
@@ -64,7 +63,13 @@ def set(self):
         if self._set or self._closed:
             return
         self._set = True
-        os.write(self._wfd, b"*")
+        try:
+            os.write(self._wfd, b"*")
+        except OSError as e:
+            if e.errno == 9 and self._closed:
```

Comment:
Is it possible for this `if`condition to ever be `True`?

We've already trapped for `self._closed == True` with the first `if` in the method, so it seems like we'd only ever hit this point with `self._closed == False`.

Reply:
Aha, this is the crux of the race condition, got it.

For the benefit of future readers of the code, please add a comment here describing the race condition as you just described it.

Otherwise we risk someone thinking the same thing I did and removing this.

Reply:
done